jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/UserTaskWithSecurityTest.java

```diff
@@ -0,0 +1,203 @@
+        File pom = new File("target/kmodule", "pom.xml");
+        pom.getParentFile().mkdir();
+        try {
+            FileOutputStream fs = new FileOutputStream(pom);
+            fs.write(getPom(releaseId).getBytes());
+            fs.close();
+        } catch (Exception e) {
+            
+        }
+		KieMavenRepository repository = getKieMavenRepository();
+        repository.deployArtifact(releaseId, kJar1, pom);
+        
+        assertNotNull(deploymentService);
+        
+        deploymentUnit = new KModuleDeploymentUnit(GROUP_ID, ARTIFACT_ID, VERSION);
+        
+        deploymentService.deploy(deploymentUnit);
+        units.add(deploymentUnit);
+    	assertNotNull(processService);
+
+    	identityProvider.setRoles(Arrays.asList("employees"));
+    }
+    
+    @After
+    public void cleanup() {
+    	if (processInstanceId != null) {
+    		try {
+		    	// let's abort process instance to leave the system in clear state
+		    	processService.abortProcessInstance(processInstanceId);
+		    	
+		    	ProcessInstance pi = processService.getProcessInstance(processInstanceId);    	
+		    	assertNull(pi);
+    		} catch (ProcessInstanceNotFoundException e) {
+    			// ignore it as it was already completed/aborted
+    		}
+    	}
+        cleanupSingletonSessionId();
+        if (units != null && !units.isEmpty()) {
+            for (DeploymentUnit unit : units) {
+            	try {
+                deploymentService.undeploy(unit);
+            	} catch (Exception e) {
+            		// do nothing in case of some failed tests to avoid next test to fail as well
+            	}
+            }
+            units.clear();
+        }
+        close();
+        CountDownListenerFactory.clear();
+    }
+    
+    
+ 
+    @Test
+    public void testGetAndWorkOnUserTasks() {
+        
+        // let's grant managers role so process can be started
+        List<String> roles = new ArrayList<String>();
+        roles.add("managers");
+        roles.add("employees");
+        identityProvider.setName("salaboy");
+        identityProvider.setRoles(roles);
+        
+        Collection<ProcessInstanceDesc> instances = runtimeDataService.getProcessInstances(new QueryContext());
+        assertNotNull(instances);
+        assertEquals(0, instances.size());
+        
+        processInstanceId = processService.startProcess(deploymentUnit.getIdentifier(), "org.jbpm.writedocument");
+        assertNotNull(processInstanceId);
+        
+        roles.clear();
+        identityProvider.setRoles(roles);
+        List<TaskSummary> tinstances = runtimeDataService.getTasksAssignedAsPotentialOwner(identityProvider.getName(), new QueryFilter());
+        assertNotNull(tinstances);
+        assertEquals(1, tinstances.size());
+        
+        Long taskID = tinstances.get(0).getId();
+        
+        userTaskService.completeAutoProgress(taskID, identityProvider.getName(), null);
```

Comment:
Shouldn't roles be added before this call? If not, a SecurityException is thrown. 
Just moving this line, 4 lines below makes it pass.